### PR TITLE
feat: add conditional custom theme with persistence and defaults.json for default export list

### LINKED ISSUE #28: Extract default list into json file
It'd be useful for folks to be able to deploy their own custom versions of the reader with default lists for their communities.

Right now they'd need to fuss with the code, but maybe we can make it easier by having them modify a single json (or newline separated text?) file.

- [ ] Decide on file format for defaults
- [ ] Add file to repo
- [ ] Load from file when detecting that the db has no follows instead of the hardcoded list
- [ ] Add docs to README on how to deploy with custom defaults.

### PR DESCRIPTION
**Pull request by akhileshthite**
closes https://github.com/hyphacoop/reader.distributed.press/issues/28

Screen Shot 2024-09-20 at 11 49 09 AM

### CODE REVIEW

File: about.html [hunk 1]
```diff
@@ -78,6 +78,17 @@ <h2>FAQs</h2>
           broader Fediverse allow for genuine reach and engagement.
         </p>
       </details>
+      <details>
```
Comment (akhileshthite) [inline about.html:81]: Screen Shot 2024-09-20 at 3 09 43 PM

File: timeline.js [hunk 2]
```diff
@@ -23,9 +26,17 @@ class ReaderTimeline extends HTMLElement {
     this.loadMoreBtnWrapper.appendChild(this.loadMoreBtn)
 
     this.loadMoreBtn.addEventListener('click', () => this.loadMore())
+
+    // Create the loading text element
+    this.loadingText = document.createElement('div')
+    this.loadingText.textContent = 'Loading...'
```
Comment (akhileshthite) [inline timeline.js:32]: Screen Shot 2024-09-20 at 3 05 28 PM

File: defaults.js [hunk 3]
```diff
@@ -21,21 +23,68 @@ export async function applyDefaults () {
       }
     }
 
-    // Apply theme if available
-    if (defaults.theme) {
-      applyTheme(defaults.theme)
+    // Determine if the custom theme differs from the default light theme
```
Comment (akhileshthite) [inline defaults.js:37]: The "custom" option will appear as a default option if the social reader default json list changes.

File: sidebar.html [hunk 4]
```diff
@@ -10,4 +10,6 @@ <h1 id="header-branding" class="header-branding">Social Reader</h1>
   <a href="https://hypha.coop/dripline/announcing-dp-social-inbox/">Social Inbox</a>
   <a href="https://distributed.press">Distributed Press</a>
 </nav>
-<distributed-search></distributed-search>
\ No newline at end of file
+<distributed-search></distributed-search>
+
+<div id="release-version" class="release-version"></div>
```
Comment (akhileshthite) [inline sidebar.html:15]: PHOTO-2024-09-29-18-31-54

**Review by RangerMauve — CHANGES_REQUESTED** [review]
THis looks great! just one comment about the version

**Comment by akhileshthite** [comment]
> THis looks great! just one comment about the version

I don't see your review comment 🤔

File: sidebar.js [hunk 5]
```diff
@@ -1,22 +1,22 @@
 import './search.js'
 import { applyDefaults } from './defaults.js'
 
-// GitHub API URL for fetching the latest release
-const githubApiUrl = 'https://api.github.com/repos/hyphacoop/reader.distributed.press/releases/latest'
+// The version will now be fetched from the default.json file
+const defaultJsonUrl = './config/defaults.json'
```
Comment (RangerMauve) [inline sidebar.js]: Does this work on all pages? I think we need absolute URLs or to use `import.meta.url` to generate a URL relative to the sidebar file

**Review by RangerMauve — CHANGES_REQUESTED** [review]

**Comment by akhileshthite** [comment]
I just tested everything. Before we merge, we need to fix one issue:

Previously, the "Load More" button would appear after following an actor, allowing me to view all their posts (main branch's timeline). However, I am now encountering an issue where the "Load More" button initially appears, but disappears from the timeline after following an actor. I am unable to navigate all the way to the oldest posts on other timelines because the "Load More" button only appears once, and doesn't reappear for subsequent loading like it used to.

**Comment by RangerMauve** [comment]
Are you getting errors?

**Comment by akhileshthite** [comment]
> Are you getting errors?

It's working now, can you test it?

**Review by RangerMauve — APPROVED** [review]
Looks good to me 👍👍👍👍

**Pull request merged**